src/EventSubscriber/PurgeQueueCommitSubscriber.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\helfi_proxy\EventSubscriber;
+
+use Drupal\purge\Plugin\Purge\Queue\QueueServiceInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Event subscriber for the purge queue commit event.
+ */
+final class PurgeQueueCommitSubscriber implements EventSubscriberInterface {
+
+  /**
+   * Constructs a PurgeQueueCommitSubscriber object.
+   */
+  public function __construct(
+    private readonly QueueServiceInterface $purgeQueue,
+  ) {}
+
+  /**
+   * Handles the purge commit event.
+   */
+  public function onPurgeQueueCommit(): void {
+    if (!method_exists($this->purgeQueue, 'commit')) {
+      // The commit method is mostly used in tests. Check if the method exists
+      // before invoking it.
+      throw new \LogicException('QueueServiceInterface::commit() does not exist anymore.');
+    }
+    $this->purgeQueue->commit();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getSubscribedEvents(): array {
+    // @todo Replace with a EVENT_NAME constant from helfi_api_base event.
+    return [
+      'helfi_cache_tag_invalidate' => ['onPurgeQueueCommit'],
```

Comment:
This can be replaced with `CacheTagInvalidatorEvent::class`

Reply:
Wouldn't the tests fail since that class is not yet committed to develop? https://github.com/City-of-Helsinki/drupal-module-helfi-api-base/pull/205

Reply:
Yeah, the tests failed since the event class is not yet in the api base source code. https://github.com/City-of-Helsinki/drupal-module-helfi-proxy/actions/runs/13763046685/job/38483225313?pr=79#step:10:73

I added there a todo task to replace it with the event class constant later.